### PR TITLE
Business rules - added exception messages and bumped carbon-analytics version

### PR DESCRIPTION
**Pull request by AnuGayan**

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java [hunk 1]
```diff
@@ -72,8 +72,7 @@ public static JsonObject fileToJson(File jsonFile) {
             Reader reader = new FileReader(jsonFile);
             jsonObject = gson.fromJson(reader, JsonObject.class);
         } catch (FileNotFoundException e) {
-            //log.error("FileNotFound Exception occurred when converting JSON file to JSON Object", e); //todo: FileNotFound exception occured. error message?
-            log.error(e.getMessage(),e);
+            log.error(e.getMessage(), e);
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: give more context and close the reader

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java [hunk 2]
```diff
@@ -172,17 +171,26 @@ public static BusinessRuleFromScratch jsonToBusinessRuleFromScratch(String jsonD
...
             }
             for (RuleTemplate ruleTemplate : templateGroup.getRuleTemplates()) {
                 validateRuleTemplate(ruleTemplate);
             }
-        } catch (TemplateManagerException x) {
+        } catch (TemplateManagerException e) {
             // System.out.println("TemplateGroup Not Valid");
-            // todo: implement
+            // todo:
+            log.error(e.getMessage(), e);
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: Give context please

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java [hunk 3]
```diff
@@ -204,31 +212,44 @@ public static void validateTemplateGroup(TemplateGroup templateGroup) throws Tem
      * @throws TemplateManagerException
      */
     public static void validateRuleTemplate(RuleTemplate ruleTemplate) throws TemplateManagerException {
-        ArrayList<String> validTemplateTypes = new ArrayList<String>(Arrays.asList(TemplateManagerConstants.SIDDHI_APP_TEMPLATE_TYPE, "gadget", "dashboard")); //todo: more types might come
-
-        if (ruleTemplate == null) {
-            // todo: throw exception
+        ArrayList<String> validTemplateTypes = new ArrayList<String>(Arrays.asList(TemplateManagerConstants
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: Use generic List here

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java [hunk 4]
```diff
@@ -172,17 +171,26 @@ public static BusinessRuleFromScratch jsonToBusinessRuleFromScratch(String jsonD
     public static void validateTemplateGroup(TemplateGroup templateGroup) throws TemplateManagerException {
         try { // todo: remove this. This is just temporary
             if (templateGroup.getName() == null) {
-                throw new TemplateManagerException("Invalid TemplateGroup configuration file found");
+                throw new TemplateManagerException("Invalid TemplateGroup configuration file found - TemplateGroup " +
+                        "name  is null" +
```
Comment (minudika) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: This has to be properly formatted

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java [hunk 5]
```diff
@@ -250,44 +271,51 @@ public static void validateRuleTemplateProperty(RuleTemplateProperty ruleTemplat
...
+                    // Find Templated Element's Name
+                    Pattern templatedElementNamePattern = Pattern.compile(TemplateManagerConstants.TEMPLATED_ELEMENT_NAME_REGEX_PATTERN);
+                    Matcher templatedElementNameMatcher = templatedElementNamePattern.matcher(templatedElement);
 
-                    templatedElements.add(templatedElementName);
+                    // When the Templated Element's Name is found
+                    if (templatedElementNameMatcher.find()) {
+                        // Templated Element's Name
+                        String templatedElementName = templatedElementNameMatcher.group(1);
+
```
Comment (TanyaM) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: unnecessary new line

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java (hunk 1 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: This is fixed by 3674cdddd78974cf7b8aca6a5feb11440c7c499b

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java [hunk 6]
```diff
@@ -250,44 +271,51 @@ public static void validateRuleTemplateProperty(RuleTemplateProperty ruleTemplat
...
             }
 
         }
 
         // All templated elements are not given in properties
+        // TODO: 9/19/17 if no script
         if (!properties.keySet().containsAll(templatedElements)) {
-            // todo: throw exception
+            throw new TemplateManagerException("All templated elements are not defined in properties");
+            // TODO: 9/19/17 pass the not implemented template field as well
```
Comment (ksdperera) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: Please remove TODO comments

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java (hunk 2 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: Fixed by 3674cdddd78974cf7b8aca6a5feb11440c7c499b

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java (hunk 3 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: Removed by 3674cdddd78974cf7b8aca6a5feb11440c7c499b

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java (hunk 5 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: This method was removed by 3674cdddd78974cf7b8aca6a5feb11440c7c499b

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java (hunk 6 above)
Comment (AnuGayan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java]: Removed by 3674cdddd78974cf7b8aca6a5feb11440c7c499b

**Pull request merged**